api/envoy/config/core/v3/address.proto

```diff
@@ -115,6 +116,12 @@ message TcpKeepalive {
   google.protobuf.UInt32Value keepalive_interval = 3;
 }
 
+message ExtraSourceAddress {
+  // The additional address to bind.
+  SocketAddress source_address = 1 [(validate.rules).message = {required: true}];
```

Comment:
nit: s/source_address/address, since it's not redundant inside the message.

Reply:
got it